### PR TITLE
Add missing QASM definitions existing as gates

### PR DESCRIPTION
**Pull request by arunraja-hub**
### Summary
Issue #4823 - Adding  missing OpenQASM definitions to qelib.ic file

### Details and comments
Currently added definition for iswap and dcx gates based on definitions in https://github.com/Qiskit/qiskit-terra/tree/master/qiskit/circuit/library/standard_gates
 More to come

### CODE REVIEW

File: qiskit/qasm/libs/qelib1.inc [hunk 1]
```diff
@@ -70,13 +70,28 @@ gate ccx a,b,c
   cx a,b; t a; tdg b;
   cx a,b;
 }
+gate dcx a, b 
+{ 
+  cx a, b; 
+  cx a, b;
```
Comment (Cryoris) [inline qiskit/qasm/libs/qelib1.inc:76]: This should be `cx b, a;` could you also fix this in dcx.py?

**Comment by 1ucian0** [comment]
I suggest to extend `test_loading_all_qelib1_gates` in `test/python/circuit/test_circuit_load_from_qasm.py` with these an other gates added in #4082, and before...

File: qiskit/circuit/library/standard_gates/dcx.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def __init__(self):
 
     def _define(self):
         """
-        gate dcx a, b { cx a, b; cx a, b; }
+        gate dcx a, b { cx b, a; cx b, a; }
```
Comment (Cryoris) [inline qiskit/circuit/library/standard_gates/dcx.py:55]:
```suggestion
        gate dcx a, b { cx a, b; cx b, a; }
```

File: qiskit/qasm/libs/qelib1.inc [hunk 3]
```diff
@@ -70,13 +70,28 @@ gate ccx a,b,c
   cx a,b; t a; tdg b;
   cx a,b;
 }
+gate dcx a, b 
+{ 
+  cx b, a;
```
Comment (Cryoris) [inline qiskit/qasm/libs/qelib1.inc:75]:
```suggestion
  cx a, b; 
```

**Comment by 1ucian0** [comment]
I know it's a bit for this comment... shouldn't additions to `qelib` increase the number  at the end? `qelib2.inc`? I always assumed the number was the version.

**Comment by Cryoris** [comment]
That's a good question -- although we added a lot of gates to `qelib1.inc` already. Maybe we should either remove the number or change to a new filename.

**Comment by 1ucian0** [comment]
ping arunraja-hub  ?

**Comment by 1ucian0** [comment]
ping @arunraja-hub ?

**Comment by 1ucian0** [comment]
We decided that QASM should be moved out of Terra. And, with it, qelib.inc too. Therefore, I'm closing this PR and tagging the issue #4823 as `on hold`.

**Pull request closed without merging**